entities/weapon.ts

```diff
@@ -0,0 +1,22 @@
+export interface DamageStatus {
+    headDamage: number
+    bodyDamage: number
+    legDamage: number
+}
+export interface WeaponStats {
+    fireRate: number
+    magazineSize: number
+    reloadTimeSeconds: number
+}
+export interface ShopData {
+    category: string
+    cost: number 
+}
+export interface Weapon {
+    uuid: string
+    displayName: string
+    displayIcon: string
+    shopData?: ShopData
+    weaponStats: WeaponStats
```

Comment:
```suggestion
    weaponStats?: WeaponStats
```

This one can be null too when the weapon is melee.

Reply:
Why did you resolve the conversation if you did not make the change?